week1/Afshin Week1.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # ex1
+
+name = input ("Enter your name :")
+surname = input ("Enter your surname :")
+string_print = f"Your name is {name} and your family is {surname }."
+print (string_print)
+
+
+# ex2
+i = 1
+x = ""
+while i < 6:
+    film = input("Enter name of your favorite film :")
+    x = film + ',' + x + '  '
+    print(x)
+    i += 1
+
+
+
+
+
+# ex3
+
+def printfirstlast():
+  color_list = ["Red","Green","White" ,"Black"]
+  print(color_list[0])
+  print(color_list[-1])
+
+
+
+
+printfirstlast()
+
+
+# ex4
+
+mydictis = {
+  0: 10,
+  1: 20
+}
+
+
+mydictis[2]=30
+
+
+mydictis
+
+
+# ex5
+
+def body():
+    weight = input ("Enter weight (kg) :")
+    height = input ("Enter weight (m) :")
+    bmi = float(weight) / float(height)**2
+    print(bmi)
+
+
+body()
+
+
+
+
+
+
+# ex6
+
+
+i = 1
+while i == 1:
+  inputNumber = input ("Your guess :")
+  print(inputNumber)
+  if inputNumber == "6":
+    print("Well guessed!")
+    break
+#  inputNumber = input ("Your guess :")
+
+
+# ex6 again
+
+
+def guessNum():
+    i= 1
+    while i == 1:
+        inputNumber = input ("Your guess :")
+        print(inputNumber)
+        if inputNumber == "6":
+            print("Well guessed!")
+            break
+        else:
+            print("Try again")
+            continue
+
+
+
+
+
+guessNum()
+
+
+# ex7
+
+
+mytuple = tuple((54, "BMW", 123, "apple", True,False))
+print(mytuple)
+
+
+# ex8
+
+
+myList = ["Isfahan", "Tehran", "Tabriz","Mashhad","Karaj"]
+M = len(myList)
+tupleX = tuple()
+for x in myList:
+    tupleX +=  x,
+    print(tupleX)
+
+
+
+
+
+
+
+# ex9
+
+
+a = [10,20,30,20,10,50,60,40,80,50,40]
+newList = []
+for x in a:
+    if len(newList) == 0:
+        newList.append(x)
+
+    else:
+        b = 0
+        for y in newList:
+            b = b + 1
+            if y == x:
+                b = b - 1
+            if y != x and b == len(newList):
+                newList.append(x)
+
+print(newList)
+
+
+# ex10
+name = input("enter your text :")
+
+
+print(len(name))
+
+i = 0
+x = ""
+while i < len(name):
+    if(i%2) == 0:
+        print (name[i])
+        x = x + name[i]
+        i += 1
+    else:
+        i += 1
+        continue
+print(x)
+
```

Comment:
Python can do this more elegant or for a _Pythonista_, there is a more _pythonic_ way of doing this.

In Python **a string is basically a list of characters**, or in other words, in Python like a list also a string is **iterable**,
and a list can be looped through like this:
```python
sample_list = [100,200,300]
for sample_int in sample_list:
    print(sample_int)
```
Do you see that there is no indexing `sample_list[i]` necessary anymore? You can do the same with a string

Now, you might have also seen that we lost the index number `i` within the loop. Yet, there is also a pythonic way of getting a running number into your for loop, see here:
```python
sample_list = [100,200,300]
for i, sample_int in enumerate(sample_list):
    print(i, sample_int)
```

Can you use my feedback and improve your code?